bin/main.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+'''Copyright (C) 2017 Collabora
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+Authors: Shane Fagan - shane.fagan@collabora.com
+'''
+
+import datetime
+import sys
+import os
+from time import sleep
+from threading import Thread
+sys.path.append(os.path.dirname(__file__))
+RULES_PATH = os.path.abspath(os.path.join('..', 'rules', 'simple'))
+
+
+class Main():
+    '''
+        Class to handle states
+    '''
+    def __init__(self):
+        pass
+
+
+def __log_message(message):
+    '''
+        Internal logging for debug
+    '''
+    with open("messages.log", "a", encoding="utf-8") as f:
+        message = "{} %% {}\n".format(datetime.datetime.now(), message)
+        f.write(message)
+        f.flush()
+        print(message)
+
+
+def run():
+    '''
+        Loop to grab logic from stdin, then spawn a threads for processing
+    '''
+    for line in sys.stdin:
+        line = line.replace(" ", "").strip('\n')
+        if line == "":
+            # Ignore blank lines
+            pass
+        if line == "quit":
+            break
+        elif "=" in line:
+            line = line.split("=")
+            Thread(target=process, args=(line[0], line[1])).start()
```

Comment:
I see no reason to complicate this with threads. Threads might be useful if we were doing I/O in 'process', Python can deal with that neatly using multiple threads. However, process is in fact a python function, it will run Python code, so it will not run in parallel with other instances of it. And worse: you need to lock the state when running it, which you currently don't do. Just dump threads, we do not need them now and probably won't need them ever.